tests/test_matprod.py

```diff
@@ -36,7 +45,31 @@ def test_matprod(nfeed, antpairs, precision, method):
     else:
         antpairs = None
 
-    obj = cls(nchunks=1, nfeed=nfeed, nant=nant, antpairs=antpairs, precision=precision)
+    if matsets and method.startswith("CPU"):
+        matsets = [
+            (np.array([0, 1, 2, 3]), np.array([0, 1, 2, 3])),
+            (np.array([0, 1, 2, 3]), np.array([3, 4])),
+            (np.array([3, 4]), np.array([0, 1, 2, 3])),
+            (np.array([3, 4]), np.array([3, 4])),
+        ]
+    elif matsets and method.startswith("GPU"):
+        matsets = [
+            (cp.array([0, 1, 2, 3]), cp.array([0, 1, 2, 3])),
+            (cp.array([0, 1, 2, 3]), cp.array([3, 4])),
+            (cp.array([3, 4]), cp.array([0, 1, 2, 3])),
+            (cp.array([3, 4]), cp.array([3, 4])),
```

Comment:
instead of making them cupy arrays here, convert them in the matprod class respectively.

Reply:
I think I figured out how to do this, but I had to use the HAVE_GPU flag instead of use_gpu to specify this, which would cause a problem if the code detected a GPU but was running in CPU mode (I don't know if this would happen).